pd-client/client.go

```diff
@@ -43,77 +50,271 @@ type Client interface {
 	Close()
 }
 
+type tsoRequest struct {
+	done     chan error
+	physical int64
+	logical  int64
+}
+
+const (
+	pdTimeout             = 3 * time.Second
+	maxMergeTSORequests   = 10000
+	maxInitClusterRetries = 100
+)
+
 type client struct {
-	worker *rpcWorker
+	urls        []string
+	clusterID   uint64
+	tsoRequests chan *tsoRequest
+
+	mu            sync.RWMutex
+	clients       map[string]pdpb2.PDClient
+	leader        string
+	checkLeaderCh chan struct{}
+
+	wg   sync.WaitGroup
+	quit chan struct{}
```

Comment:
I meant Context is not recommended to be used like that. See https://github.com/golang/go/blob/master/src/context/context.go#L29-L31